### PR TITLE
Add data layout strategy generator app

### PR DESCRIPTION
**Pull request by teamurko**
## Summary

This is part 4 of a new feature data layout optimization.

Added Spark app wrapper. This PR builds on top of https://github.com/linkedin/openhouse/pull/116

The following 3 components will be added eventually:
DLO library that has primitives for generating data layout optimization strategies (done)
App that generates strategies for all tables (this PR)
Scheduling of the app
## Changes

- [ ] Client-facing API Changes
- [ ] Internal API Changes
- [ ] Bug Fixes
- [x] New Features
- [ ] Performance Improvements
- [ ] Code Style
- [x] Refactoring
- [ ] Documentation
- [ ] Tests

For all the boxes checked, please include additional details of the changes made in this pull request.  

## Testing Done

- [ ] Manually Tested on local docker setup. Please include commands ran, and their output.
- [ ] Added new tests for the changes made.
- [ ] Updated existing tests to reflect the changes made.
- [ ] No tests added or updated. Please explain why. If unsure, please feel free to ask for help.
- [ ] Some other form of testing like staging or soak time in production. Please explain.

For all the boxes checked, include a detailed description of the testing done for the changes made in this pull request.
The wrapped code is part of an existing integration test.

# Additional Information

- [ ] Breaking Changes
- [ ] Deprecations
- [ ] Large PR broken into smaller PRs, and PR plan linked in the description.

For all the boxes checked, include additional details of the changes made in this pull request.

### CODE REVIEW

File: apps/spark/src/main/java/com/linkedin/openhouse/jobs/spark/RewriteStrategyGeneratorSparkApp.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.cli.CommandLine;
+import org.apache.commons.cli.Option;
+import org.apache.spark.sql.SparkSession;
+
+@Slf4j
+public class RewriteStrategyGeneratorSparkApp extends BaseTableSparkApp {
```
Comment (sumedhsakdeo) [inline apps/spark/src/main/java/com/linkedin/openhouse/jobs/spark/RewriteStrategyGeneratorSparkApp.java:19]: DataLayoutStrategyGeneratorSparkApp

**Comment by sumedhsakdeo** [comment]
Curious on the renames from DataLayout to RewriteStrategy

File: apps/spark/src/main/java/com/linkedin/openhouse/jobs/spark/RewriteStrategyGeneratorSparkApp.java (hunk 1 above)
Comment (teamurko) [reply apps/spark/src/main/java/com/linkedin/openhouse/jobs/spark/RewriteStrategyGeneratorSparkApp.java:19]: It was long `DataLayoutOptimizationStrategyGeneratorSparkApp` :) `DataLayoutStrategy` is missing action like rewrite, or optimization. What would you suggest, do you want `DataLayout` to be part of the name?

Comment (sumedhsakdeo) [reply apps/spark/src/main/java/com/linkedin/openhouse/jobs/spark/RewriteStrategyGeneratorSparkApp.java:19]: Yeah rewrite strategy is too generic

Comment (sumedhsakdeo) [reply apps/spark/src/main/java/com/linkedin/openhouse/jobs/spark/RewriteStrategyGeneratorSparkApp.java:19]: Fine with DataLayoutStrategyGenerator

**Review by sumedhsakdeo — APPROVED** [review]

**Review by sumedhsakdeo — APPROVED** [review]

**Pull request merged**